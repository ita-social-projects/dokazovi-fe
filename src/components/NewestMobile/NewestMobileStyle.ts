import { makeStyles } from '@material-ui/core';
import { Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) => ({
  root: {
    paddingTop: theme.spacing(2),
    backgroundColor: theme.palette.background.default,
  },
  selected: {
    color: 'black !important',
    fontWeight: 700,
  },
  textColorPrimary: {
    color: theme.palette.text.secondary,
  },
  loader: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: 80,
  },
  content: {
    marginTop: theme.spacing(5),
  },
}));
