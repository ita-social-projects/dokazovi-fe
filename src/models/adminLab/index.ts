/* eslint-disable import/no-cycle */

export {
  adminLabReducer,
  setStateToInit,
  setSort,
  setFilter,
  setPage,
  setFiltersToInit,
  setField,
} from './reducers';
export { getMaterialsAction, archiveAdminPost } from './asyncActions';
export { selectAdminLab, selectMeta } from './selectors';
