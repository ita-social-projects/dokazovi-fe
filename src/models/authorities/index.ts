/* eslint-disable */
export { getAuthoritiesAsyncAction } from './asyncAction';
export { authoritiesReducer } from './reducers';
export { selectAuthorities } from './selectors';
