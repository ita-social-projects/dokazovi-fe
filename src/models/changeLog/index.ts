/* eslint-disable import/no-cycle */
export { changeLogReducer, setChangesSize } from './reducer';
export { selectSize } from './selectors';
