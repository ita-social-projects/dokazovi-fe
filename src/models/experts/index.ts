/* eslint-disable import/no-cycle */
export {
  fetchExperts,
  fetchExpertById,
  fetchExpertMaterials,
} from './asyncActions';
export { resetMaterials, loadPosts, expertsReducer } from './reducers';
export {
  selectExperts,
  selectLoadingExpertsPosts,
  selectLoadingExperts,
  selectExpertsData,
} from './selectors';
