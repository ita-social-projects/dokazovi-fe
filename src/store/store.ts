import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import rootReducer from './rootReducer';
import type { RootStateType } from './rootReducer';

export const store = configureStore({
  reducer: rootReducer,
});

export type AppThunkType = ThunkAction<
  void,
  RootStateType,
  unknown,
  Action<string>
>;

export type AppDispatchType = typeof store.dispatch;
