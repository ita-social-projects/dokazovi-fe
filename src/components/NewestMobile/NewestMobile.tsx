import React, { useEffect, useState } from 'react';
import SwipeableViews from 'react-swipeable-views';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { CircularProgress } from '@material-ui/core';
import { langTokens } from '../../locales/localizationInit';
import { useStyles } from './NewestMobileStyle';
import { useActions } from '../../shared/hooks';
import {
  fetchNewestMobile,
  selectMobileMaterials,
} from '../../models/newestPostsMobile';
import { AutoPaginationPostList } from '../AutoPaginationPostList/AutoPaginationPostList';

const a11yProps = (index: number) => {
  return {
    id: `full-width-tab-${index}`,
    'aria-controls': `full-width-tabpanel-${index}`,
  };
};

export const NewestMobile: React.FC = () => {
  const { t } = useTranslation();

  const classes = useStyles();

  const [value, setValue] = useState(0);

  const [boundFetchMobileMaterials] = useActions([fetchNewestMobile]);
  const content = useSelector(selectMobileMaterials);

  useEffect(() => {
    boundFetchMobileMaterials();
  }, []);

  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue);
  };

  const handleChangeIndex = (index: number) => {
    setValue(index);
  };

  return (
    <>
      <div>
        <AppBar position="static" color="default">
          <Tabs
            value={value}
            onChange={handleChange}
            indicatorColor="primary"
            textColor="primary"
            variant="fullWidth"
            aria-label="full width tabs example"
          >
            <Tab
              classes={classes}
              label={t(langTokens.experts.expertOpinion_1)}
              {...a11yProps(0)}
            />
            <Tab
              classes={classes}
              label={t(langTokens.common.translation)}
              {...a11yProps(1)}
            />
            <Tab
              classes={classes}
              label={t(langTokens.common.media)}
              {...a11yProps(2)}
            />
            <Tab
              classes={classes}
              label={t(langTokens.common.video)}
              {...a11yProps(3)}
            />
          </Tabs>
        </AppBar>
        {content.length ? (
          <SwipeableViews
            axis="x"
            index={value}
            onChangeIndex={handleChangeIndex}
            className={classes.content}
          >
            <div
              role="tabpanel"
              id="full-width-tabpanel-0"
              aria-labelledby="full-width-tab-0"
            >
              <AutoPaginationPostList
                posts={content[0]?.postDTOS && content[0].postDTOS}
              />
            </div>
            <div
              role="tabpanel"
              id="full-width-tabpanel-1"
              aria-labelledby="full-width-tab-1"
            >
              <AutoPaginationPostList
                posts={content[1]?.postDTOS && content[1].postDTOS}
              />
            </div>
            <div
              role="tabpanel"
              id="full-width-tabpanel-2"
              aria-labelledby="full-width-tab-2"
            >
              <AutoPaginationPostList
                posts={content[2]?.postDTOS && content[2].postDTOS}
              />
            </div>
            <div
              role="tabpanel"
              id="full-width-tabpanel-3"
              aria-labelledby="full-width-tab-3"
            >
              <AutoPaginationPostList
                posts={content[3]?.postDTOS && content[3].postDTOS}
              />
            </div>
          </SwipeableViews>
        ) : (
          <div className={classes.loader}>
            <CircularProgress />
          </div>
        )}
      </div>
    </>
  );
};
