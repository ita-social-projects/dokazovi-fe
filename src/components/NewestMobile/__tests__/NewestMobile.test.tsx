import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { MemoryRouter, Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from 'models/store';

import { mockMobileContent } from './mockMobileContent';
import { NewestMobile } from '../NewestMobile';

store.getState = () => mockMobileContent;

const history = createMemoryHistory();

describe('NewestContainer component renders correctly', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <NewestMobile />
        </MemoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('Component renders with correct default view', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <NewestMobile />
        </MemoryRouter>
      </Provider>,
    );
  });

  it('There are four clickable names of the parts of the Newest section on the Main Page', () => {
    expect(screen.getAllByRole('tab')).toHaveLength(4);
  });

  it('User is redirected to the "Думки експертів" part by default', () => {
    expect(screen.getByText('ТЕСТ publish')).toBeInTheDocument();
  });

  it('Only one part of the Newest section is demonstrated at once', () => {
    expect(screen.getAllByText('Думка експерта')).toHaveLength(3);
    expect(screen.queryByText('ТЕСТТЕСТТЕСТТЕСТТЕСТ')).toBeNull();
  });
});

describe('User can change the demonstrated part', () => {
  beforeEach(() => {
    const mockVisualViewport = {
      pageTop: 585,
    };

    Object.defineProperty(window, 'visualViewport', {
      writable: true,
      configurable: true,
      value: mockVisualViewport,
    });

    render(
      <Provider store={store}>
        <MemoryRouter>
          <NewestMobile />
        </MemoryRouter>
      </Provider>,
    );
  });

  it('User can choose the demonstrated part by clicking on the translation tab', () => {
    const translationTab = screen.getByTestId('full-width-tab-1');
    fireEvent.click(translationTab);
    expect(translationTab).toHaveAttribute('aria-selected', 'true');
    expect(screen.getByText('Єва Євенко')).toBeInTheDocument();
    expect(screen.getByTestId('full-width-tab-0')).toHaveAttribute(
      'aria-selected',
      'false',
    );
  });

  it('User can choose the demonstrated part by clicking on the media tab', () => {
    const mediaTab = screen.getByTestId('full-width-tab-2');
    fireEvent.click(mediaTab);
    expect(mediaTab).toHaveAttribute('aria-selected', 'true');
    expect(screen.getByText('ТЕСТТЕСТТЕСТТЕСТТЕСТ')).toBeInTheDocument();
    expect(screen.getByTestId('full-width-tab-0')).toHaveAttribute(
      'aria-selected',
      'false',
    );
  });

  it('User can choose the demonstrated part by clicking on the video tab', () => {
    const videoTab = screen.getByTestId('full-width-tab-3');
    fireEvent.click(videoTab);
    expect(videoTab).toHaveAttribute('aria-selected', 'true');
    expect(
      screen.getByText('erertewrtwertwertwertwertwertwertwertwertwert'),
    ).toBeInTheDocument();
    expect(screen.getByTestId('full-width-tab-0')).toHaveAttribute(
      'aria-selected',
      'false',
    );
  });
});

describe("Material page should be opened by clicking on the material title or material card's text", () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router history={history}>
          <NewestMobile />
        </Router>
      </Provider>,
    );
  });

  it("clicking on materials card redirects on it's page", () => {
    const links = screen.getAllByText('Думка експерта');
    const link = links[0].closest('a');
    fireEvent.click(link);
    expect(history.location.pathname).toBe('/posts/117');
  });
});

describe("By clicking on button 'See all' should be redirected to page with all materials", () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router history={history}>
          <NewestMobile />
        </Router>
      </Provider>,
    );
  });

  it("clicking on 'Всі думки експертів' redirects to page with all materials of this type", () => {
    const link = screen.getByText('Всі думки експертів').closest('button');
    fireEvent.click(link);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?origins=1');
  });

  it("clicking on 'Всі переклади' redirects to page with all materials of this type", () => {
    fireEvent.click(screen.getByTestId('full-width-tab-1'));
    const link = screen.getByText('Всі переклади').closest('button');
    fireEvent.click(link);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?origins=3');
  });

  it("clicking on 'Всі медитеки' redirects to page with all materials of this type", () => {
    fireEvent.click(screen.getByTestId('full-width-tab-2'));
    const link = screen.getByText('Всі медитеки').closest('button');
    fireEvent.click(link);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?origins=2');
  });

  it("clicking on 'Всі відео' redirects to page with all materials of this type", () => {
    fireEvent.click(screen.getByTestId('full-width-tab-3'));
    const link = screen.getByText('Всі відео').closest('button');
    fireEvent.click(link);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?origins=0&types=2');
  });
});
