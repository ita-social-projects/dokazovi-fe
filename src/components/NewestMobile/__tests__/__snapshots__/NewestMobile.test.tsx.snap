// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewestContainer component renders correctly should match snapshot 1`] = `
<div>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root makeStyles-appBarRoot-3 MuiAppBar-positionFixed MuiAppBar-colorDefault makeStyles-stickyTop-11 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="makeStyles-container-15"
      >
        <div
          class="MuiTabs-root makeStyles-buttonsRoot-4"
        >
          <div
            class="MuiTabs-scroller makeStyles-scroller-16 MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              aria-label="full width tabs example"
              class="MuiTabs-flexContainer makeStyles-tabsContainer-14"
              role="tablist"
            >
              <button
                aria-controls="full-width-tabpanel-0"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root makeStyles-tabRoot-8 MuiTab-textColorPrimary Mui-selected makeStyles-selected-5 MuiTab-fullWidth"
                data-testid="full-width-tab-0"
                id="full-width-tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper makeStyles-wrapper-9"
                >
                  Думки експертів
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="full-width-tabpanel-1"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root makeStyles-tabRoot-8 MuiTab-textColorPrimary MuiTab-fullWidth"
                data-testid="full-width-tab-1"
                id="full-width-tab-1"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiTab-wrapper makeStyles-wrapper-9"
                >
                  Переклади
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="full-width-tabpanel-2"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root makeStyles-tabRoot-8 MuiTab-textColorPrimary MuiTab-fullWidth"
                data-testid="full-width-tab-2"
                id="full-width-tab-2"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiTab-wrapper makeStyles-wrapper-9"
                >
                  Медитека
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="full-width-tabpanel-3"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root makeStyles-tabRoot-8 MuiTab-textColorPrimary MuiTab-fullWidth"
                data-testid="full-width-tab-3"
                id="full-width-tab-3"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiTab-wrapper makeStyles-wrapper-9"
                >
                  Відео
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-17 PrivateTabIndicator-colorSecondary-19 MuiTabs-indicator makeStyles-indicator-2"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </header>
    <div
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
      >
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            aria-labelledby="full-width-tab-0"
            id="full-width-tabpanel-0"
            role="tabpanel"
          >
            <div
              class="PostsList-masonryGrid-21"
            >
              <div
                class="PostsList-masonryColumn-22"
                style="width: 50%;"
              >
                <div
                  class="PostsList-masonryItem-23"
                >
                  <div
                    class="MuiPaper-root MuiCard-root ExpertOpinionPostPreviewCard-root-24 MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <a
                      href="/experts/13"
                      target="_blank"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-33 ExpertOpinionPostPreviewCard-header-27"
                      >
                        <div
                          class="MuiBox-root MuiBox-root-34"
                        >
                          <div
                            class="MuiBox-root MuiBox-root-35"
                          >
                            <a
                              href="/posts/117"
                            >
                              <span
                                class="MuiTypography-root ExpertOpinionPostPreviewCard-postType-31 MuiTypography-overline"
                              >
                                Думка експерта
                              </span>
                            </a>
                            <span
                              class="MuiTypography-root ExpertOpinionPostPreviewCard-authorFullName-26 MuiTypography-h5"
                            >
                              Марина Вовк
                            </span>
                          </div>
                          <div
                            class="MuiCardMedia-root ExpertOpinionPostPreviewCard-avatar-25"
                            data-testid="avatar"
                            style="background-image: url(https://i.pravatar.cc/300?img=5);"
                            title="Марина Вовк"
                          />
                        </div>
                        <span
                          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                        >
                          Lorem ipsum dolor sit amet, consectetur adipiscing
                        </span>
                      </div>
                    </a>
                    <a
                      href="/posts/117"
                      target="_blank"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-36 ExpertOpinionPostPreviewCard-body-28"
                      >
                        <div
                          class="MuiCardContent-root ExpertOpinionPostPreviewCard-content-32"
                        >
                          <h3
                            class="MuiTypography-root ExpertOpinionPostPreviewCard-textHeader-29 MuiTypography-h4 MuiTypography-gutterBottom"
                          >
                            ТЕСТ publish
                          </h3>
                          <p
                            class="MuiTypography-root ExpertOpinionPostPreviewCard-textBody-30 MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
                          >
                            Fermentum iaculis eu non diam phasellus vestibulum lorem sed.
                          </p>
                        </div>
                        <div
                          class="MuiBox-root MuiBox-root-37"
                        >
                          <span
                            class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
                          >
                            16 січня
                          </span>
                        </div>
                      </div>
                    </a>
                  </div>
                </div>
                <div
                  class="PostsList-masonryItem-23"
                >
                  <div
                    class="MuiPaper-root MuiCard-root ExpertOpinionPostPreviewCard-root-24 MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <a
                      href="/experts/14"
                      target="_blank"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-48 ExpertOpinionPostPreviewCard-header-27"
                      >
                        <div
                          class="MuiBox-root MuiBox-root-49"
                        >
                          <div
                            class="MuiBox-root MuiBox-root-50"
                          >
                            <a
                              href="/posts/114"
                            >
                              <span
                                class="MuiTypography-root ExpertOpinionPostPreviewCard-postType-31 MuiTypography-overline"
                              >
                                Думка експерта
                              </span>
                            </a>
                            <span
                              class="MuiTypography-root ExpertOpinionPostPreviewCard-authorFullName-26 MuiTypography-h5"
                            >
                              Олег Петренко
                            </span>
                          </div>
                          <div
                            class="MuiCardMedia-root ExpertOpinionPostPreviewCard-avatar-25"
                            data-testid="avatar"
                            style="background-image: url(https://i.pravatar.cc/300?img=11);"
                            title="Олег Петренко"
                          />
                        </div>
                        <span
                          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                        >
                          Lorem ipsum dolor sit amet, consectetur adipiscing
                        </span>
                      </div>
                    </a>
                    <a
                      href="/posts/114"
                      target="_blank"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-51 ExpertOpinionPostPreviewCard-body-28"
                      >
                        <div
                          class="MuiCardContent-root ExpertOpinionPostPreviewCard-content-32"
                        >
                          <h3
                            class="MuiTypography-root ExpertOpinionPostPreviewCard-textHeader-29 MuiTypography-h4 MuiTypography-gutterBottom"
                          >
                            Допис Lorem Ipsum
                          </h3>
                          <p
                            class="MuiTypography-root ExpertOpinionPostPreviewCard-textBody-30 MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
                          >
                            Існує багато варіацій уривків з Lorem Ipsum, але більшість з них зазнала певних змін на кшталт жартівливих вставок або змішування слів, які навіть не виглядають правдоподібно.
                          </p>
                        </div>
                        <div
                          class="MuiBox-root MuiBox-root-52"
                        >
                          <span
                            class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
                          >
                            22 жовтня
                          </span>
                        </div>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="PostsList-masonryColumn-22"
                style="width: 50%;"
              >
                <div
                  class="PostsList-masonryItem-23"
                >
                  <div
                    class="MuiPaper-root MuiCard-root ExpertOpinionPostPreviewCard-root-24 MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <a
                      href="/experts/10"
                      target="_blank"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-38 ExpertOpinionPostPreviewCard-header-27"
                      >
                        <div
                          class="MuiBox-root MuiBox-root-39"
                        >
                          <div
                            class="MuiBox-root MuiBox-root-40"
                          >
                            <a
                              href="/posts/116"
                            >
                              <span
                                class="MuiTypography-root ExpertOpinionPostPreviewCard-postType-31 MuiTypography-overline"
                              >
                                Думка експерта
                              </span>
                            </a>
                            <span
                              class="MuiTypography-root ExpertOpinionPostPreviewCard-authorFullName-26 MuiTypography-h5"
                            >
                              Марія Марієнко
                            </span>
                          </div>
                          <div
                            class="MuiCardMedia-root ExpertOpinionPostPreviewCard-avatar-25"
                            data-testid="avatar"
                            style="background-image: url(https://i.pravatar.cc/300?img=16);"
                            title="Марія Марієнко"
                          />
                        </div>
                        <span
                          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                        >
                          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat duis.
                        </span>
                      </div>
                    </a>
                    <a
                      href="/posts/116"
                      target="_blank"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-41 ExpertOpinionPostPreviewCard-body-28"
                      >
                        <div
                          class="MuiCardContent-root ExpertOpinionPostPreviewCard-content-32"
                        >
                          <h3
                            class="MuiTypography-root ExpertOpinionPostPreviewCard-textHeader-29 MuiTypography-h4 MuiTypography-gutterBottom"
                          >
                            This is title
                          </h3>
                          <p
                            class="MuiTypography-root ExpertOpinionPostPreviewCard-textBody-30 MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
                          >
                            Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet
                          </p>
                        </div>
                        <div
                          class="MuiBox-root MuiBox-root-42"
                        >
                          <span
                            class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
                          >
                            14 січня
                          </span>
                        </div>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-buttonContainer-10"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-13"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Всі думки експертів
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
